src/awkward/operations/ak_linear_fit.py

```diff
@@ -76,22 +76,28 @@ def linear_fit(x, y, weight=None, axis=None, *, keepdims=False, mask_identity=Fa
 def _impl(x, y, weight, axis, keepdims, mask_identity):
     axis = regularize_axis(axis)
     behavior = behavior_of(x, y, weight)
+    backend = backend_of(x, y, weight, coerce_to_common=True)
     x = ak.highlevel.Array(
-        ak.operations.to_layout(x, allow_record=False, allow_other=False),
+        ak.operations.to_layout(x, allow_record=False, allow_other=False).to_backend(
+            backend
+        ),
```

Comment:
Previously we could have had mixed backends until the ufuncs later in this routine. It is better to convert up-front.

Reply:
:heavy_check_mark: